app/models/specific_tube_creation.rb

```diff
@@ -10,13 +12,30 @@ class ChildPurpose < ApplicationRecord # rubocop:todo Style/Documentation
 
   validates :child_purposes, presence: true
 
+  has_many :parent_associations,
+           foreign_key: 'asset_creation_id',
+           class_name: 'AssetCreation::ParentAssociation',
+           inverse_of: 'asset_creation'
+  has_many :parents, through: :parent_associations, class_name: 'Labware'
+
   # [Array<Hash>] An optional array of hashes which get passed in to the create! action
   #               on tube_purpose.
   #               Allows overriding default attributes, or setting custom
   #               values for. eg. name.
   #               eg. [{ name: 'Tube one' }, { name: 'Tube two' }]
   attr_writer :tube_attributes
 
+  def set_parents=(uuids)
+    self.parents = uuids.map { |uuid| Uuid.find_by(external_id: uuid).resource }
+    self.parent = parents.first
+  end
+
+  def set_parent=(uuid)
+    parent = Uuid.find_by(external_id: uuid).resource
+    self.parents = [parent]
+    self.parent = parent
+  end
```

Comment:
I must admit, I thought the API handled some of this automatically if the relationships were defined.  Ignoring the parent association for the moment (As that's a little more complicated) What happens if you change:
```
set_parents <= parents
```
To just
```
parents <=> parents
```

Reply:
I know I have had to resort to tricks like this before though.

Reply:
Yeah, I tried a few different combinations and couldn't find a simpler one that worked. Maybe we could step through it together on Tues in case there are some blockers you could help me past?

Reply:
I think with your suggestion for the `def parent` method, it wouldn't support existing 'write' actions through the API that use `parent` rather than `parents` ? I was trying to keep it backwards compatible.

Reply:
Resolved with help from James - the issue was that `parents <=> parents` triggers an exception in the core api code - has to be write only (`parents <= parents`)